File: src/main/java/com/dumbdogdiner/warrior/commands/arena/ArenaJoinCommand.java
package com.dumbdogdiner.warrior.commands.arena;

import com.dumbdogdiner.warrior.Warrior;
import com.dumbdogdiner.warrior.api.user.WarriorUser;
import com.dumbdogdiner.warrior.api.arena.Arena;
import com.dumbdogdiner.warrior.api.command.SubCommand;
import com.dumbdogdiner.warrior.api.sessions.ArenaSession;
import com.dumbdogdiner.warrior.api.translation.Constants;
import com.dumbdogdiner.warrior.managers.ArenaManager;
import com.dumbdogdiner.warrior.managers.PlayerManager;
import com.dumbdogdiner.warrior.utils.TranslationUtil;
import org.bukkit.Sound;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;

public class ArenaJoinCommand implements SubCommand {
    @Override
    public String getAlias() {
        return "join";
    }

    @Override
    public String getSyntax() {
        return "/arena join <Arena>";
    }

    @Override
    public String getPermission() {
        return "warrior.arena.join";
    }

    @Override
    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
        if(args.length < 2 || args[1].length() > 18) {
            return false;
        }

        WarriorUser user = PlayerManager.get(((Player)sender).getUniqueId());
        Arena a = ArenaManager.get(args[1]);

        if(a == null) {
            String msg = Warrior.getTranslator().translate(Constants.Lang.ARENA_DOESNT_EXIST, new HashMap<>() {
                {
                    put("ARENA", args[1]);
                }
            });
            user.sendMessage(TranslationUtil.getPrefix() + msg);
            user.playSound(Sound.ENTITY_ITEM_BREAK, 0.5f, 1f);
            return true;
        }

        new BukkitRunnable() {

            @Override
            public void run() {
                user.setSession(new ArenaSession(user.getUserId(), a));
                String msg = Warrior.getTranslator().translate(Constants.Lang.ARENA_TELEPORT, new HashMap<String, String>() {
                    {
                        put("ARENA", a.getName());
                    }
                });

                user.sendMessage(TranslationUtil.getPrefix() + msg);
                user.playSound(Sound.ENTITY_ENDERMAN_TELEPORT, 0.5f, 1f);
            }

        }.runTask(Warrior.getInstance());

        return true;
    }

    @Override
    public List<String> getArguments(CommandSender sender, String[] arguments) {
        List<String> strings = new ArrayList<>();

        if(arguments.length == 2) strings = ArenaManager.getArenas().stream().map(Arena::getName).collect(Collectors.toList());

        return strings;

    }
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/commands/arena/ArenaJoinCommand.kt
package com.dumbdogdiner.warrior.commands.arena

import com.dumbdogdiner.warrior.api.command.SubCommand
import com.dumbdogdiner.warrior.api.arena.Arena
import com.dumbdogdiner.warrior.Warrior
import java.util.HashMap
import com.dumbdogdiner.warrior.api.util.TranslationUtil
import com.dumbdogdiner.warrior.api.sessions.ArenaSession
import com.dumbdogdiner.warrior.api.translation.Constants
import com.dumbdogdiner.warrior.user.User
import org.bukkit.Sound
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.bukkit.scheduler.BukkitRunnable
import java.util.ArrayList
import java.util.function.Function
import java.util.stream.Collectors

class ArenaJoinCommand : SubCommand {
    override fun getAlias(): String {
        return "join"
    }

    override fun getSyntax(): String {
        return "/arena join <Arena>"
    }

    override fun getPermission(): String {
        return "warrior.arena.join"
    }

    override fun execute(sender: CommandSender, commandLabel: String, args: Array<String>): Boolean {
        if (args.size < 2 || args[1].length > 18) {
            return false
        }
        val user: User = Warrior.userCache[(sender as Player).uniqueId]
        val a: Arena? = Warrior.arenaManager[args[1]]
        if (a == null) {
            val msg: String =
                Warrior.translator.translate(Constants.Lang.ARENA_DOESNT_EXIST, object : HashMap<String, String>() {
                    init {
                        put("arena", args[1])
                    }
                }, user)
            user.sendMessage(TranslationUtil.getPrefix() + msg)
            user.playSound(Sound.ENTITY_ITEM_BREAK, 0.5f, 1f)
            return true
        }
        object : BukkitRunnable() {
            override fun run() {
                user.setSession(ArenaSession(user.uniqueId, a))
                val msg: String = Warrior.translator.translate(Constants.Lang.ARENA_TELEPORT, object : HashMap<String?, String?>() {
                        init {
                            put("arena", a.name)
                        }
                    }, user)
                user.sendMessage(TranslationUtil.getPrefix() + msg)
                user.playSound(Sound.ENTITY_ENDERMAN_TELEPORT, 0.5f, 1f)
            }
        }.runTask(Warrior.instance)
        return true
    }

    override fun getArguments(sender: CommandSender, arguments: Array<String>): List<String> {
        var strings: List<String> = ArrayList()
        if (arguments.size == 2) strings =
            Warrior.arenaManager.arenas.stream().map<String>(Function { obj: Arena -> obj.name })
                .collect<List<String>, Any>(Collectors.toList<String>())
        return strings
    }
}